bindings/netstandard/ElectionGuard/ElectionGuard.Decryption/DecryptionMediator.cs

```diff
@@ -219,24 +198,96 @@ List<CiphertextBallot> ballots
         }
     }
 
-    protected override void DisposeUnmanaged()
+    #endregion
+
+    #region Accumulate Tally
+
+    // accumulate can be called before CreateChallenge
+    // but it doesnt have to be because CreateChallenge will also handle it
+    public void AccumulateShares(string tallyId, bool skipValidation = false)
```

Comment:
you can call accumulateshares directly (like, in the background) or not. if you dont call it, the CiphertextTallyDecryption will run it prior to creating the challenge if it needs to.